trieste/acquisition/function/continuous_thompson_sampling.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2021 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains acquisition function builders for continuous Thompson sampling.
+"""
+from __future__ import annotations
+
+from typing import Optional
+
+import tensorflow as tf
+
+from ...data import Dataset
+from ...models import ProbabilisticModel
+from ...models.interfaces import TrajectoryFunction
+from ...types import TensorType
+from ..interface import SingleModelGreedyAcquisitionBuilder
+
+
+class GreedyContinuousThompsonSampling(SingleModelGreedyAcquisitionBuilder[ProbabilisticModel]):
+    r"""
+
+    Acquisition function builder for performing greedy continuous Thompson sampling. This builder
+    return acquisition functions that are the negatives of approximate samples from the
+    given :class:`ProbabilisticModel`, as provided by the model's :meth:`get_negative_trajectory`
+    method. A set of such samples are to be maximized in a sequential greedy manner to provide
+    the next recommended query points.
+
+    For more details about trajectory-based Thompson sampling see :cite:`hernandez2017parallel` and
+    :cite:`wilson2020efficiently`.
+    """
+
+    def prepare_acquisition_function(
+        self,
+        model: ProbabilisticModel,
+        dataset: Optional[Dataset] = None,
+        pending_points: Optional[TensorType] = None,
+    ) -> TrajectoryFunction:
+        """
+        :param model: The model.
+        :param dataset: The data from the observer (not used).
+        :param pending_points: The points already in the current batch (not used).
+        :return: A trajectory sampled from the model.
+        """
+
+        try:
+            self._trajectory_sampler = model.trajectory_sampler()
```

Comment:
guess you don't fancy defining a `HasTrajectorySampler` protocol (and ideally also `HasReparamSampler`)?

Reply:
done